exporter/opentelemetry-exporter-datadog/src/opentelemetry/exporter/datadog/exporter.py

```diff
@@ -312,19 +312,23 @@ def _parse_tags_str(tags_str):
     return parsed_tags
 
 
-def _extract_tags_from_resource(resource):
+def _extract_tags_from_resource(resource, fallback_service_name):
     """Parse tags from resource.attributes, except service.name which
     has special significance within datadog"""
     tags = {}
-    service_name = None
     if not (resource and getattr(resource, "attributes", None)):
-        return [tags, service_name]
+        return [tags, fallback_service_name]
```

Comment:
Technically `tags` could just be `{}` here since it's not really use here at this point too! But that's not in scope for your PR I would say :-)

Reply:
good catch. As you can tell my code is not particularly pythonic (i'm mostly ruby these days 😅 ). But there's a lot in this little exporter that could get cleaned up. I'll make a note to also try to give this code another pass for https://github.com/open-telemetry/opentelemetry-python-contrib/issues/574